e2e/tests/eip712_test.go

```diff
@@ -0,0 +1,58 @@
+package tests
+
+import (
+	"context"
+	"math"
+	"testing"
+
+	"github.com/bnb-chain/greenfield/e2e/core"
+	storageutils "github.com/bnb-chain/greenfield/testutil/storage"
+	storagetypes "github.com/bnb-chain/greenfield/x/storage/types"
+	"github.com/stretchr/testify/suite"
+)
+
+type Eip712TestSuite struct {
+	core.BaseSuite
+}
+
+func (s *Eip712TestSuite) SetupSuite() {
+	s.BaseSuite.SetupSuite()
+}
+
+func (s *Eip712TestSuite) SetupTest() {
+}
+
+func TestEip712TestSuite(t *testing.T) {
+	suite.Run(t, new(Eip712TestSuite))
+}
+
+func (s *Eip712TestSuite) TestMultiMessages() {
+	var err error
+	sp := s.StorageProviders[0]
+	user := s.GenAndChargeAccounts(1, 1000000)[0]
+
+	// CreateBucket
+	bucketName := storageutils.GenRandomBucketName()
+	msgCreateBucket := storagetypes.NewMsgCreateBucket(
+		user.GetAddr(), bucketName, storagetypes.VISIBILITY_TYPE_PUBLIC_READ, sp.OperatorKey.GetAddr(),
+		nil, math.MaxUint, nil, 0)
+	msgCreateBucket.PrimarySpApproval.Sig, err = sp.ApprovalKey.Sign(msgCreateBucket.GetApprovalBytes())
+	s.Require().NoError(err)
+
+	// UpdateBucketInfo
```

Comment:
@flywukong We can use this feature to add an `CreateBucketIfNotExist` option to CreateObjectOptions which means create the bucket if it does not exist. This will bring better user exprience.